src/dymaptic.GeoBlazor.Core/Components/MapComponent.razor.cs

```diff
@@ -240,6 +271,421 @@ protected virtual async Task RenderView(bool forceRender = false)
+    /// </param>
+    /// <param name="once">
+    ///     Whether to fire the callback only once.
+    /// </param>
+    /// <param name="initial">
+    ///     Whether to fire the callback immediately after initialization, if the necessary conditions are met.
+    /// </param>
+    /// <typeparam name="T">
+    ///     The type of return value to expect in the handler.
+    /// </typeparam>
+    /// <exception cref="UnMatchedTargetNameException">
+    ///     This exception is thrown when a watchExpression's target object name doesn't match the targetName parameter.
+    /// </exception>
+    /// <remarks>
+    ///     For adding watchers to types other than <see cref="MapView"/> and <see cref="SceneView"/>, the default <code>targetName</code> should not be relied upon. Make sure it matches the variable in your <code>watchExpression</code>
+    /// </remarks>
+    public Task AddReactiveWatcher<T>(string watchExpression, Func<T, Task> handler, string? targetName = null,
+        bool once = false, bool initial = false)
+    {
+        return AddReactiveWatcherImplementation(watchExpression, handler, targetName, once, initial);
+    }
+    
+    /// <summary>
+    ///     Tracks any properties accessed in the <code>watchExpression</code> and calls the callback when any of them change.
+    /// </summary>
+    /// <param name="watchExpression">
+    ///     Expression used to get the current value. All accessed properties will be tracked.
+    /// </param>
+    /// <param name="handler">
+    ///     The function to call when there are changes.
+    /// </param>
+    /// <param name="targetName">
+    ///     The name of the target you are referencing in the <code>watchExpression</code>. For example, if the expression is "layer?.refresh", then the targetName should be "layer". The type of the target should also match the class on which this method was called.
+    /// </param>
+    /// <param name="once">
+    ///     Whether to fire the callback only once.
+    /// </param>
+    /// <param name="initial">
+    ///     Whether to fire the callback immediately after initialization, if the necessary conditions are met.
+    /// </param>
+    /// <typeparam name="T">
+    ///     The type of return value to expect in the handler.
+    /// </typeparam>
+    /// <exception cref="UnMatchedTargetNameException">
+    ///     This exception is thrown when a watchExpression's target object name doesn't match the targetName parameter.
+    /// </exception>
+    /// <remarks>
+    ///     For adding watchers to types other than <see cref="MapView"/> and <see cref="SceneView"/>, the default <code>targetName</code> should not be relied upon. Make sure it matches the variable in your <code>watchExpression</code>
+    /// </remarks>
+    // ReSharper disable once UnusedMethodReturnValue.Global
+    public Task AddReactiveWatcher<T>(string watchExpression, Action<T> handler, string? targetName = null, 
```

Comment:
We talked about this the other day, but I'm worried about how to know what my options are for this `T`. Just knowing some basic things like how we map classes from the JS namespace to the GeoBlazor namespace would help in the events doc.

If I recall, using `<string>` will cause it to return the JSON string of the object to use myself, which is great! Are there any conflicts between that JSON representation and something that actually returns a string?

Reply:
I don't think there is a conflict, string will handle both raw strings and json. I agree about the docs. After I fix the other event handlers to return event types, we can document all the possibilities for each event, or at least the most common.